internal/provider/store.go

```diff
@@ -73,43 +73,41 @@ func (s *ProviderStore) Inventory(data InventoryDataSourceModel) inventory.Inven
 
 // Logger initializes the context logger for the given inventory.
 func (s *ProviderStore) Logger(ctx context.Context, inv InventoryDataSourceModel, withs ...any) (context.Context, error) {
-	if s.providerResourceData.Log == nil {
-		return ctx, nil
-	}
-
-	if lf := s.providerResourceData.Log.File; lf != nil {
-		ihash, err := s.Encode(inv.Seed.ValueString())
-		if err != nil {
-			return ctx, fmt.Errorf("failed to encode inventory hash: %w", err)
-		}
-		logpath := fmt.Sprintf("%s.log", ihash)
+	if s.providerResourceData.Log != nil {
```

Comment:
can we early return when `s.providerResourceData.log == nil && s.providerResourceData.Log.File == nil`

rather than have a double nested if clause?

Reply:
restructured things to bail early if there's no matching configs from the provider 🙏